controller/metrics/metrics.go

```diff
@@ -243,6 +254,15 @@ func (m *MetricsServer) IncSync(app *argoappv1.Application, state *argoappv1.Ope
 	m.syncCounter.WithLabelValues(app.Namespace, app.Name, app.Spec.GetProject(), app.Spec.Destination.Server, string(state.Phase), strconv.FormatBool(isDryRun)).Inc()
 }
 
+// IncAppSyncDuration observes app sync duration
+func (m *MetricsServer) IncAppSyncDuration(app *argoappv1.Application, state *argoappv1.OperationState) {
+	if !state.Phase.Successful() {
+		return
+	}
+	m.syncDuration.WithLabelValues(app.Namespace, app.Name, app.Spec.GetProject(), app.Spec.Destination.Server).
```

Comment:
`app.Spec.Destination.Server` should be received in parameters bacuse it can be `nil` if destination by name is used